app/controllers/hiring_staff/vacancies/job_location_controller.rb

```diff
@@ -0,0 +1,76 @@
+class HiringStaff::Vacancies::JobLocationController < HiringStaff::Vacancies::ApplicationController
+  include FirstStepFormConcerns
+
+  before_action :redirect_unless_school_group_user_flag_on
+  before_action :set_up_url
+  before_action only: %i[create update] do
+    set_up_form(JobLocationForm)
+  end
+
+  def show
+    if @vacancy.present?
+      @form = JobLocationForm.new(@vacancy.attributes)
+    elsif session[:vacancy_attributes].present?
+      @form = JobLocationForm.new(session[:vacancy_attributes])
+    else
+      @form = JobLocationForm.new(school_group_id: current_school_group.id)
+    end
+  end
+
+  def create
+    store_vacancy_attributes(@form.vacancy.attributes)
+
+    if @form.complete_and_valid?
+      session_vacancy_id ? update_vacancy(form_params) : save_vacancy_without_validation
+      store_vacancy_attributes(@form.vacancy.attributes)
+      return redirect_to_next_step_if_continue(@vacancy&.id.present? ? @vacancy.id : session_vacancy_id)
```

Comment:
I think we're moving to a standard approach of `if... render, else... render` so let's change when we see this more pythonic 🐍  style